### DISCUSSION TITLE
[SYCL-MLIR] How can I run sycl-mlir on Nvidia GPU

### DISCUSSION THREAD

**Discussion by SealSoul117** [intel/llvm#14567] [Q&A]
Hi! I'm tring to use sycl-mlir on Nvidia A100 to test my stencil program. 

It seems the only way to compile with sycl-mlir is to use

`clang++ -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir example.cpp`

However, the program compiled this way is for the SPIR-V backend, and it will throw `PI_ERROR_INVALID_BINARY` error when I excute the binary file.

And if my target in  `-fsycl-targets` is `nvptx64-nvidia-cuda`, it seems it's using DPC++ instead of sycl-mlir.

So, I want to ask:

1. How can I get a excutable file compiled by sycl-mlir on Nvidia GPU?
2. If a program is compiled for SPIR-V backend, is it excuatble only on Intel platforms?

**Comment by AlexeySachkov** [comment]
For the first question I will tag @etiotto here.

> If a program is compiled for SPIR-V backend, is it excuatble only on Intel platforms?

It should be executable on any platform that supports SPIR-V, i.e. the list is not limited by Intel platforms theoretically, but in practice, it is not that huge due to a few caveats:
- Use of any SYCL extension will lead to use of corresponding SPIR-V extensions which may not be supported on non-Intel platforms
- We may set quite high requirements for a backend in terms of SPIR-V version supported: I think that our toolchain currently can emit SPIR-V up to version 1.4

**Comment by etiotto** [comment]
> 1. How can I get a excutable file compiled by sycl-mlir on Nvidia GPU?

Compilation for NVidia GPUs has not been considered.

**Reply by SealSoul117** [reply]
So currently, it is only available on Intel GPUs?

**Reply by etiotto** [reply]
> So currently, it is only available on Intel GPUs?

Correct. This is also highly experimental work and currently not maintained.

**Reply by SealSoul117** [reply]
Thank you very much. May I ask little more? What's the the reason sycl-mlir is only available on Intel GPUs? Is it because of some extensions to SPIR-V as AlexeySachkov mentioned above?

**Reply by SealSoul117** [reply]
And, if I want to add some optimizations for stencil to it, where could I start?
